testrunner/src/mill/testrunner/TestRunnerUtils.scala

```diff
@@ -213,6 +228,111 @@ import scala.jdk.CollectionConverters.IteratorHasAsScala
     (doneMessage, results.toSeq)
   }
 
+  private def stealTaskFromSelectorFolder(
+      testClasses: Seq[(Class[?], Fingerprint)],
+      runner: Runner,
+      stealFolder: os.Path,
+      selectorFolder: os.Path
+  ): Array[Task] = {
+    while (true) {
+      val files = os.list(selectorFolder)
+      if (files.nonEmpty) {
+        val offset = Random.nextInt(files.size)
+        val stole =
+          try {
+            os.move(
+              files(offset),
+              stealFolder / s"selector-stolen",
+              atomicMove = true
+            )
+            true
+          } catch {
+            case e: Exception => false
+          }
+        if (stole) {
+          val selectors = os.read.lines(stealFolder / s"selector-stolen")
+          val classFilter = TestRunnerUtils.globFilter(selectors)
+          val tasks = runner.tasks(
+            for ((cls, fingerprint) <- testClasses.iterator.toArray if classFilter(cls.getName))
+              yield new TaskDef(
+                cls.getName.stripSuffix("$"),
+                fingerprint,
+                false,
+                Array(new SuiteSelector)
+              )
+          )
+          os.remove(stealFolder / s"selector-stolen")
```

Comment:
Let's leave the files on disk once stolen, no need to delete them. That would help debugging since people can `ls` each worker's baseFolder and see what test classes that worker claimed

Reply:
We can reuse the original file name to avoid conflicts, rather than renaming them to `selector-stolen` as you have done here.